create_admin_user.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# Usage create_admin_user.sh -u user -p password -h host -k private_key
+#
+# Description of required command line arguments:
+#
+#   -u: admin user name
+#   -p: admin user password
+#   -h: host name or IP address
+#   -k: path to private SSH key file
+#
+# Usage example: create_admin_user.sh -u xrd -p secret -h host -k private_key_file
+
+usage() {
+  echo "Usage: create_admin_user.sh -u xrd -p secret -h host -k private_key_file"
+}
+
+exit_abnormal() {
+  usage
+  exit 1
+}
+
+create_user() {
+    user=$1
+    pass=$2
+    host=$3
+    ssh_key=$4
+    os_name=$(ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "grep '^NAME' /etc/os-release")
+
+    # create user and add to sudo/root group
+    if echo "$os_name" | grep 'Ubuntu'; then
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "useradd $user --group sudo -p $pass"
+    fi
+    if echo "$os_name" | grep 'Centos'; then
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "useradd --system --home /var/lib/xroad --no-create-home --shell /bin/bash --user-group --comment \"X-Road system user\" xroad"
```

Comment:
This ``useradd`` seems fragile here, just two failure cases:

1. security server software has ALREADY been installed on Centos, the ``xroad`` user already exists, command will  fail.
2. security server software has NOT been installed on Centos. In such case the user will get **non-existant** home directory.

Reply:
It is now checked if these procedures have already been performed or not, so this should be ok